src/aind_hcr_data_transformation/compress/czi_to_zarr.py

```diff
@@ -5,467 +5,318 @@
+    if bucket_name is not None:
+        kvstore_dict = {
+            "driver": "s3",
+            "bucket": bucket_name,
+            "aws_region": aws_region,
+            # "aws_credentials": {
+            #     "type": "profile",
+            #     "profile": aws_profile,
+            #     "credentials_file": os.path.expanduser(credentials_file),
+            # },
+            "context": {
+                "cache_pool": {"total_bytes_limit": read_cache_bytes},
+                "data_copy_concurrency": {"limit": cpu_cnt},
+                "s3_request_concurrency": {"limit": cpu_cnt},
+                "experimental_s3_rate_limiter": {
+                    "read_rate": cpu_cnt,
+                    "write_rate": cpu_cnt,
+                },
+            },
+        }
 
-    Returns
-    -------
-    A list of dictionaries for each axis
-    """
-    axes_5d = [
-        {"name": "t", "type": "time", "unit": f"{time_unit}"},
-        {"name": "c", "type": "channel"},
-        {"name": "z", "type": "space", "unit": f"{space_unit}"},
-        {"name": "y", "type": "space", "unit": f"{space_unit}"},
-        {"name": "x", "type": "space", "unit": f"{space_unit}"},
-    ]
-    return axes_5d
+    return {
+        "driver": "zarr3",
+        "kvstore": {
+            **kvstore_dict,
+            "path": output_path,
+        },
+        "path": str(scale),
+        "recheck_cached_metadata": False,
+        "recheck_cached_data": False,
+        "metadata": {
+            "shape": data_shape,
+            "zarr_format": 3,
+            "node_type": "array",
+            "fill_value": 0,
+            "chunk_grid": {
+                "name": "regular",
+                "configuration": {"chunk_shape": shard_shape},
+            },
+            "chunk_key_encoding": {
+                "name": "default",
+                "configuration": {"separator": "/"},
+            },
+            "attributes": {
+                "dimension_units": zyx_resolution,
+            },
+            "dimension_names": ["t", "c", "z", "y", "x"],
+            "codecs": [
+                {
+                    "name": "sharding_indexed",
+                    "configuration": {
+                        "chunk_shape": chunk_shape,
+                        "codecs": [
+                            {
+                                "name": "bytes",
+                                "configuration": {"endian": "little"},
+                            },
+                            {
+                                "name": "blosc",
+                                "configuration": compressor_kwargs,
+                            },
+                        ],
+                        "index_codecs": [
+                            {
+                                "name": "bytes",
+                                "configuration": {"endian": "little"},
+                            },
+                            {"name": "crc32c"},
```

Comment:
Great to see a checksum codec here. Is there support for a 16 bit version?

Reply:
Yes, but in theory this codec it should work with any dtype because the bytes codecs are for the bytes themselves and how they are stored (e.g., little endian).